batch/batch.go

```diff
@@ -1404,14 +1421,22 @@ func (b *Batch) makeFragments(frags, clearFrags fragments) (fragments, fragments
 		opts := field.Options
 		curShard := ^uint64(0) // impossible sentinel value for shard.
 		var curBM *roaring.Bitmap
+		var existCurBM *roaring.Bitmap
 		for j := range b.ids {
 			col, rowIDs := b.ids[j], rowIDSets[j]
-			if len(rowIDs) == 0 {
-				continue
-			}
 			if col/shardWidth != curShard {
 				curShard = col / shardWidth
 				curBM = frags.GetOrCreate(curShard, fname, "")
+				if opts.TrackExistence {
+					existCurBM = frags.GetOrCreate(curShard, fname, "existence")
+				}
+			}
+			if len(rowIDs) == 0 {
+				// you can validly specify an empty set, which is not the same as a null
+				if opts.TrackExistence && !(opts.Type == featurebase.FieldTypeTime && opts.NoStandardView) && rowIDs != nil {
```

Comment:
I'm not sure. I feel like this is really logically the conjunction of a couple of things. One is a logical query for "are we, in fact, tracking existence", which is a mess to figure out; we write to the existence field if we are tracking existence, unless the field is a time field with NoStandardView, or a BSI field. But then also, we're distinguishing between "nil" and "empty set".

So this feels like... there's a real underlying question which is roughly "are we tracking existence in the existence field", and then this should be that `&& rowIDs != nil`, because that's two separate questions, that we're and-ing, but this way of writing it looks like three and a half questions. So yeah, sort of agreed.